js/sort.js

```diff
@@ -0,0 +1,59 @@
+import {getRandomPositiveInteger} from './utils.js';
+const initSubmitSort = (arr, render)=>{
+
+  const filterDiscussed = document.querySelector('#filter-discussed');
+  const filterRandom = document.querySelector('#filter-random');
+  const filterDefault = document.querySelector('#filter-default');
+  const cloneArr = arr;
+
+  const removeElement = ()=>{
+    const picture = document.querySelectorAll('.picture');
+    picture.forEach((element) => {
+      element.remove();
+    });
+  };
+
+  const sortByPopularity = () => {
+    setTimeout(() => {
+      cloneArr.sort((a, b) => b.comments.length - a.comments.length);
+      render(cloneArr);
+    }, 1000);
+  };
+  const addEventsDiscussed = (evt)=>{
+    evt.preventDefault();
+    removeElement();
+    sortByPopularity();
+  };
+
+  const sortRandomElements = ()=>{
```

Comment:
лучше поправь сортировку, добавив максимальный размер до 10, иначе будет бесконечный цикл, если вдруг фоток будет меньше 10. Само число 10 можно вынести в константу в начало файла
```suggestion
  const sortRandomElements = ()=>{
    const sortArr = [];
    const maxSize = Math.min(cloneArr.length, 10);
    while (sortArr.length !== maxSize) {
      const randomElement = cloneArr[getRandomPositiveInteger(0, cloneArr.length - 1)];
      if (!sortArr.includes(randomElement)) {
        sortArr.push(randomElement);
      }
    }
    render(sortArr);
  };
```